main.go

```diff
@@ -346,3 +354,45 @@ func makeHealthHandler() func(http.ResponseWriter, *http.Request) {
 		}
 	}
 }
+
+type concurrencyLimiter struct {
```

Comment:
I do have an interest in this feature, but I would like you to follow the process and work with the community. The of-watchdog may also not be the correct place to address this problem.